src/main/java/com/jcraft/jsch/KeyPair.java

```diff
@@ -950,11 +967,18 @@ else if (data != null &&
                   publicKeyComment = Util.byte2str(buf, start, i-start);
                 }
               } 
+          } else if (buf[0] == 'e' && buf[1] == 'c' && buf[2] == 'd' && buf[3] == 's') {
+            if (buf.length > 7) {
+                if (buf.length > 24 &&
+                        buf[6] == 's' && buf[7] == 'h' && buf[8] == 'a' && buf[9] == '2' &&
+                        buf[11] == 'n' && buf[12] == 'i' && buf[13] == 's' && buf[14] == 't' && buf[15] == 'p' &&
+                        buf[16] == '2' && buf[17] == '5' && buf[18] == '6' &&
+                        buf[20] == 'c' && buf[21] == 'e' && buf[22] == 'r' && buf[23] == 't') {
+                    type = ECDSA_CERT;
+                } else {
+                    type = ECDSA;
+                }
```

Comment:
It would be nice to maintain the existing coding style here (namely no spaces surrounding `=`, `==`, etc.).